clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -71,9 +77,36 @@ public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V
         this.key = key;
         this.value = value;
         this.timestamp = timestamp;
+        this.headers = new RecordHeaders(headers);
     }
 
     /**
+     * Creates a record with a specified timestamp to be sent to a specified topic and partition
+     *
+     * @param topic The topic the record will be appended to
+     * @param partition The partition to which the record should be sent
+     * @param timestamp The timestamp of the record
+     * @param key The key that will be included in the record
+     * @param value The record contents
+     */
+    public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V value) {
+        this(topic, partition, timestamp, key, value, null);
+    }
+
+    /**
+     * Creates a record to be sent to a specified topic and partition
+     *
+     * @param topic The topic the record will be appended to
+     * @param partition The partition to which the record should be sent
+     * @param key The key that will be included in the record
+     * @param value The record contents
+     * @param headers The headers that will be included in the record
+     */
+    public ProducerRecord(String topic, Integer partition, K key, V value,  Headers headers) {
```

Comment:
Yes, but one takes `Headers` and the other takes `Iterable<Header>` and it's unclear why that is so.